src/test/resources/lib/configFile/initIDProvider-test.js

```diff
@@ -0,0 +1,159 @@
+const test = require('/lib/xp/testing');
+
+
+
+//////////////////////  Mock workaround (test.mock only works once)
+
+const configFileMocks = {};
+test.mock("/lib/configFile/configFile.js", {
+    getAllIdProviderNames: () => configFileMocks.getAllIdProviderNames(),
+    shouldAutoInit: () => configFileMocks.shouldAutoInit(),
+    getConfigForIdProvider: (idProviderName) => configFileMocks.getConfigForIdProvider(idProviderName)
+});
+const setConfigFileMocks = mocks => {
+    Object.keys(configFileMocks).forEach( key => {
+        delete configFileMocks[key];
+    });
+    Object.keys(mocks).forEach( key => {
+        configFileMocks[key] = mocks[key];
+    })
+};
+
+const beanMocks = {};
+test.mock("/lib/configFile/services/bean.js", {
+    getIdProviders: () => beanMocks.getIdProviders(),
+    createIdProvider: (params) => beanMocks.createIdProvider(params)
+});
+const setBeanMocks = mocks => {
+    Object.keys(beanMocks).forEach( key => {
+        delete beanMocks[key];
+    });
+    Object.keys(mocks).forEach( key => {
+        beanMocks[key] = mocks[key];
+    })
+};
+
+
+
+const lib = require('./initIDProvider');
+
+
+
+/////////////////////////////////////////////
+
+exports.test_initIDProvider_initUserStores_addOnlyNewIDPs_ifShouldAutoInit = () => {
+    const CREATED_IDPS = {};
+    let addedProviders = 0;
+
+    setConfigFileMocks({
+        getAllIdProviderNames: () => ["myidp1", "myidp2", "myidp3", , "myidp4"],
+        shouldAutoInit: () => true,
+        getConfigForIdProvider: (idProviderName) => ({
+            displayName: "Mock " + idProviderName,
+            description: "Mock description",
+        })
+    });
+
+    setBeanMocks({
+        getIdProviders: () => [
+            {
+                "key": "myidp1",
+                "displayName": "Mock idp1",
+                "idProviderConfig": {
+                    "applicationKey": "com.enonic.app.oidcidprovider",
+                    "config": []
+                }
+            },
+            {
+                "key": "myidp2",
+                "displayName": "Mock idp2",
+                "idProviderConfig": {
+                    "applicationKey": "com.enonic.app.oidcidprovider",
+                    "config": []
+                }
+            }
+        ],
+        createIdProvider: (params) => {
+            CREATED_IDPS[params.name] = params;
+            addedProviders++;
+            return true;
+        }
+    });
+
+    lib.initUserStores();
+
+    // Only myidp3 and -4 should have been added, since myidp1 and -2 are already in system repo,
+    // according to the mocked getIdProviders:
+
+    test.assertEquals(2, addedProviders);
```

Comment:
Should we not do `test.assertEquals(2, getIdProviders() - 2)` Since we know two of them are created by default? 
Then we could remove all `createIdProvider` methods and `addedProviders` ? If this tests more then it should or is difficult to restructure just leave as is.

Reply:
This test is kinda twofold - both that idproviders that aren't already present are added, and that those that are present are not added.